usautobuild/config.py

```diff
@@ -29,54 +30,57 @@ class Config:
     target_platforms = ["linuxserver", "StandaloneWindows64", "StandaloneOSX", "StandaloneLinux64"]
     cdn_download_url = "https://unitystationfile.b-cdn.net/{}/{}/{}.zip"
     forkname = "UnityStationDevelop"
-    output_dir = Path(os.getcwd(), 'builds')
-    license_file = Path(os.getcwd(), "UnityLicense.ulf")
+    output_dir = Path.cwd() / "builds"
+    license_file = Path.cwd() / "UnityLicense.ulf"
     discord_webhook = ""
     abort_on_build_fail = True
 
     project_path = ""
     build_number = 0
 
-    def __init__(self, args: dict, config_file: str = None):
+    def __init__(self, args: dict, config_file: Optional[str] = None):
         self.args = args
         if config_file:
-            self.config_file = config_file
+            self.config_file = Path(config_file)
         self.handle_config_file()
         self.handle_args()
         self.check_required_envs()
         self.get_required_envs()
 
     def check_required_envs(self):
-        logger.info("Checking required envs...")
+        log.info("Checking required envs...")
         for env in self.required_envs:
-            i = os.getenv(env)
-            if i is None:
-                logger.error(f"Required env is missing: {env}")
+            if env not in os.environ:
+                log.error(f"Required env is missing: {env}")
                 raise MissingRequiredEnv(env)
 
     def get_required_envs(self):
-        self.cdn_host = os.getenv("CDN_HOST")
-        self.cdn_user = os.getenv("CDN_USER")
-        self.cdn_password = os.getenv("CDN_PASSWORD")
-        self.docker_password = os.getenv("DOCKER_PASSWORD")
-        self.docker_username = os.getenv("DOCKER_USERNAME")
-        self.changelog_api_url = os.getenv("CHANGELOG_API_URL")
-        self.changelog_api_key = os.getenv("CHANGELOG_API_KEY")
+        self.cdn_host = os.environ["CDN_HOST"]
```

Comment:
Why change getenv for environ?

Reply:
because getenv is Optional[str]. code already checked these required variables. if some are missing and return None thats's a bug

Reply:
actually this should be done in a loop too. ill change it now

Reply:
done now. also removed check function because having them separate and looping twice did not make any sense